File: MDPServer/src/main/java/com/example/MDPServer/service/ScheduleService.java
package com.example.MDPServer.service;

import com.example.MDPServer.domain.repository.ScheduleRepository;
import com.example.MDPServer.dto.ScheduleDTO;
import com.example.MDPServer.dto.UserDTO;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ScheduleService {
    private ScheduleRepository scheduleRepository;

    @Autowired
    public ScheduleService(ScheduleRepository scheduleRepository){
        this.scheduleRepository = scheduleRepository;
    }

    public JSONObject postSchedule(ScheduleDTO scheduleDTO){
        JSONObject json = new JSONObject();
        try{
            scheduleRepository.save(scheduleDTO.toEntity());
            json.put("status", "OK");
            return json;
        }catch (Exception e){
            e.printStackTrace();
            json.put("status", "FAIL");
            return json;
        }
    }

    public List<?> getScheduleList(Long userNo){
        return scheduleRepository.findAllByUserNo(UserDTO.builder().userNo(userNo).build().toEntity());
    }
    public ScheduleDTO getSchedule(Long scheduleNo){
        var schedule = scheduleRepository.findByScheduleNo(scheduleNo).get();
        ScheduleDTO scheduleDTO = ScheduleDTO.builder()
                .scheduleNo(schedule.getScheduleNo())
                .airlineId(schedule.getAirlineId())
                .depAirportName(schedule.getDepAirportName())
                .depAirportId(schedule.getDepAirportId())
                .arrAirportName(schedule.getArrAirportName())
                .arrAirportId(schedule.getArrAirportId())
                .arrPlandTime(schedule.getArrPlandTime())
                .depPlandTime(schedule.getDepPlandTime())
                .airlineName(schedule.getAirlineName())
                .vihicleId(schedule.getVihicleId())
                .userNo(UserDTO.builder().userNo(schedule.getUserNo().getUserNo()).build().toEntity())
                .build();
        return scheduleDTO;
    }
    public JSONObject deleteSchedule(Long scheduleNo){
        try{
            scheduleRepository.deleteById(scheduleNo);
            return new JSONObject().put("status", "OK");
        }catch (Exception e){
            return new JSONObject().put("status", "FAIL");
        }

    }
}

File: mdpserver2/src/service/ScheduleService.ts
import { HttpException, Injectable } from "@nestjs/common";
import { find } from "rxjs";
import { ScheduleRepository } from "src/domain/repository/ScheduleRepository";
import { CreateScheduleDTO, ScheduleDTO, SeatDTO } from "src/dto/ScheduleDto";

@Injectable()
export class ScheduleService{
    constructor(private scheduleRepository : ScheduleRepository){
        this.scheduleRepository = scheduleRepository
    }

    async getScheduleList(userNo : string) : Promise<ScheduleDTO[] | undefined | null>{
        const result = await this.scheduleRepository.findListByUserNo(userNo)
        if(result.length <= 0){
            throw new HttpException(Object.assign({
                statusCode : 404,
                message : "예약일정이 없습니다",
            }),404)
        }
        return result
    }
    async createSchedule(createScheduleDTO : CreateScheduleDTO) : Promise<void>{
        await this.scheduleRepository.save(createScheduleDTO)
        return Promise.resolve()
    }
    async deleteSchedule(scheduleNo : string) : Promise<void>{
        const find = await this.scheduleRepository.findByScheduleNo(scheduleNo)
        if(!find){
            throw new HttpException(Object.assign({
                statusCode : 404,
                message : "스케줄에 대한 정보가 없습니다",
            }),404)
        }
        await this.scheduleRepository.delete(scheduleNo)
        return Promise.resolve()
    }
    async getSchedule(schedule : string) : Promise<ScheduleDTO>{
        const find = await this.scheduleRepository.findByScheduleNo(schedule)
        if(!find){
            throw new HttpException(Object.assign({
                statusCode : 404,
                message : "스케줄에 대한 정보가 없습니다",
            }),404)
        }

        return find
    }
    
    async getSeat(vihicleId : string) : Promise<ScheduleDTO[] | undefined | null>{
        const result = await this.scheduleRepository.findByVihideId(vihicleId)
        if(!find){
            throw new HttpException(Object.assign({
                statusCode : 404,
                message : "스케줄에 대한 정보가 없습니다"
            }), 404)
        }
        return result
    }
}